src/tensor_conversion.hpp

```diff
@@ -34,4 +47,156 @@ Status convertStringRequestToOVTensor2D(const TensorType& src, ov::Tensor& tenso
+            }
+            image = std::move(imageCorrectPrecision);
+        }
+        if (!targetHeight.isStatic() || !targetWidth.isStatic()) {
+            return StatusCode::INTERNAL_ERROR;
+        }
+        if (tensor_conversion::resizeNeeded(image, targetHeight.getStaticValue(), targetWidth.getStaticValue())) {
+            if (!resizeSupported) {
+                return StatusCode::INVALID_SHAPE;
+            }
+            cv::Mat imageResized;
+            status = tensor_conversion::resizeMat(image, imageResized, targetHeight.getStaticValue(), targetWidth.getStaticValue());
+            if (!status.ok()) {
+                return status;
+            }
+            image = std::move(imageResized);
+        }
+
+        // if (i == 0 && src.contents().bytes_contents_size() > 1) {
+        //     // Multiply src.string_val_size() * image resolution * precision size
+        // }
+        images.push_back(image);
+    }
+    return StatusCode::OK;
+}
+
+template <typename TensorType>
+Status convertNativeFileFormatRequestTensorToOVTensor(const TensorType& src, ov::Tensor& tensor, const TensorInfo& tensorInfo, const std::string* buffer) {
+    OVMS_PROFILE_FUNCTION();
+    auto status = validateTensor(tensorInfo, src, buffer);
+    if (status != StatusCode::OK) {
+        SPDLOG_DEBUG("Input native file format validation failed");
+        return status;
+    }
+    std::vector<cv::Mat> images;
+    status = convertTensorToMatsMatchingTensorInfo(src, images, tensorInfo, buffer);
+    if (!status.ok()) {
+        SPDLOG_DEBUG("Input native file format conversion failed");
+        return status;
+    }
+    tensor = tensor_conversion::convertMatsToTensor(images, tensorInfo);
+    if (!tensor) {
+        SPDLOG_DEBUG("Input native file format conversion failed");
+        return StatusCode::IMAGE_PARSING_FAILED;
+    }
+    return StatusCode::OK;
+}
+
+class InferenceTensor;
+
+template <>
+Status convertNativeFileFormatRequestTensorToOVTensor(const InferenceTensor& src, ov::Tensor& tensor, const TensorInfo& tensorInfo, const std::string* buffer);
+
+template <typename TensorType>
+Status convertStringRequestToOVTensor(const TensorType& src, ov::Tensor& tensor, const std::string* buffer) {
+    OVMS_PROFILE_FUNCTION();
+    if (buffer != nullptr) {
+        return convertBinaryExtensionStringFromBufferToNativeOVTensor(src, tensor, buffer);
+    }
+    int batchSize = getBinaryInputsSize(src);
+    tensor = ov::Tensor(ov::element::Type_t::string, ov::Shape{static_cast<size_t>(batchSize)});
+    std::string* data = tensor.data<std::string>();
+    for (int i = 0; i < batchSize; i++) {
+        data[i].assign(getBinaryInput(src, i));
+    }
+    return StatusCode::OK;
+}
+
+template <>
+Status convertStringRequestToOVTensor(const InferenceTensor& src, ov::Tensor& tensor, const std::string* buffer);
+
+template <typename TensorType>
+Status convertOVTensor2DToStringResponse(const ov::Tensor& tensor, TensorType& dst) {
```

Comment:
why do we need so much in header file? it was in cpp as i understand it correctly

Reply:
Because if this is template we need to have definition available at the point of template instantiation with specific type.